validate/validate.go

```diff
@@ -377,59 +378,55 @@ func (v *Validator) CheckProcess() (errs error) {
 // CheckCapabilities checks v.spec.Process.Capabilities
 func (v *Validator) CheckCapabilities() (errs error) {
 	process := v.spec.Process
-	if v.platform == "linux" {
```

Comment:
I think we need to decide whether these platform guards belong inside the check function, outside it, or both.  For example, we currently [call `CheckRlimits` regardless of platform][1] and have [a guard inside it to no-op on platforms where it isn't defined][2].  The `if` check you're removing here is basically the same sort of no-op internal guard.  And you're *adding* a similar internal no-op guard to `CheckHooks` in this PR with 4b20ce20f.

The most robust approach would be to guard on both sides:

* Internally, raise an error if you call a check function for a platform that doesn't define the checked property (e.g. calling `CheckCapabilities` for Windows or Solaris).
* Externally, only call check functions on the appropriate platforms.

Thoughts?

[1]: https://github.com/opencontainers/runtime-tools/blob/acbe085c54698d0ccb98892efbc233f61d2faed5/validate/validate.go#L362
[2]: https://github.com/opencontainers/runtime-tools/blob/acbe085c54698d0ccb98892efbc233f61d2faed5/validate/validate.go#L440-L441

Reply:
Agree. I will update.

Reply:
updated, PTAL